ci/ansible/playbooks/os/rhel/upgrade-release.yml

```diff
@@ -0,0 +1,558 @@
+---
+# Based on https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/upgrading_from_rhel_7_to_rhel_8/index
+# and partially on https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-in-place-upgrade
+
+# This play requires a Leapp metadata archive from the Red Hat portal which cannot be shared publicly.
+# Local path to this archive must be provided via 'leapp_archive' variable, for example:
+
+# ansible-playbook -e leapp_archive=/absolute/path/leapp-data15.tar.gz
+
+# Requirements:
+# - System attached to RHUI repositories or Red Hat subscription ('non_cloud' provider)
+
+# Limitations:
+# - Ansible connection as root is not supported (PermitRootLogin)
+
+# TODO:
+# Fix issue when running on AWS on repository host deployed with epicli 1.0.2 (VM crash)
+
+- name: In-place RHEL release upgrade
+  hosts: "{{ target | default('all') }}"
+  become: true
+  vars:
+    versions:
+      required: {major: '7', full: '7.9'}  # minimal version from which upgrade is supported
+      target: {major: '8', full: '8.5'}
+
+    leapp_dependencies:
+      packages:
+        aws:
+          - leapp-rhui-aws
+          - rh-amazon-rhui-client
+        azure:
+          - leapp-rhui-azure
+          - rhui-azure-rhel7
+        non_cloud:
+          - leapp-upgrade
+      repos:
+        aws:
+          - name: rhui-client-config-server-7
+            repo_file: /etc/yum.repos.d/redhat-rhui-client-config.repo
+          - name: rhel-7-server-rhui-extras-rpms
+            repo_file: /etc/yum.repos.d/redhat-rhui.repo
+        azure:
+          - name: rhui-microsoft-azure-rhel7
+            repo_file: /etc/yum.repos.d/rh-cloud.repo
+          - name: rhui-rhel-7-server-rhui-extras-rpms
+            repo_file: /etc/yum.repos.d/rh-cloud.repo
+        non_cloud:
+          - name: rhel-7-server-extras-rpms
+            repo_file: /etc/yum.repos.d/redhat.repo
+
+    reboot:
+      connect_timeout: 5  # maximum seconds to wait for a successful connection to the managed hosts before next trial
+      update_timeout: 500
+      upgrade_timeout: 3600  # long wait required by Leapp utility service
+
+  tasks:
+    - name: End the play for already upgraded hosts
+      meta: end_host
+      when: ansible_distribution_version is version(versions.target.full, '>=')
+
+    ### PRE-UPGRADE ###
```

Comment:
Maybe this can be taken to separate file? This play is too long.

Reply:
Agree that this playbook is too long but it's meant to be used only for our CI pipelines. Having everything in one file allows us to use simple and fast `curl` call to download the playbook in the pipelines. Having multiple files we would have to use more complex curl calls like `curl -O [URL1] -O [URL2]`. Furthermore, I assume that the playbook will be rather short-life code.